### ISSUE TITLE
FR: Support for Azure AD identity types when creating a machine catalog

### ISSUE THREAD

**Issue by j7lloyd** [citrix/terraform-provider-citrix#25]
From reviewing the code for **machine_catalog_resource.go**, I appreciate that Active Directory is the only type that you currently support, hard-coded as follows:

```
var provisioningScheme citrixorchestration.CreateMachineCatalogProvisioningSchemeRequestModel
provisioningScheme.SetNumTotalMachines(int32(plan.ProvisioningScheme.NumTotalMachines.ValueInt64()))
provisioningScheme.SetIdentityType(citrixorchestration.IDENTITYTYPE_ACTIVE_DIRECTORY) // Non-Managed setup does not support non-domain joined
provisioningScheme.SetWorkGroupMachines(false)                                        // Non-Managed setup does not support non-domain joined
```

Of course, I also understand that there are other considerations to enable and configure all of the available join types listed here: https://docs.citrix.com/en-us/citrix-daas/install-configure/create-machine-identities-joined-catalogs.html. However, I was hoping (selfishly) that extended support for IDENTITYTYPE_HYBRID_AZURE_AD would be a relatively quick win, given the similarities?

**Comment by AlanCitrix** [comment]
Yes we are very interested in getting this to work. It's on the roadmap, thanks for the feedback. Will leave this issue open to track.

**Comment by zhuolun-citrix** [comment]
Hi @j7lloyd ,

We have added support for Hybrid Azure AD identity type for MCS machine catalogs in our latest update #36 . Please update your provider to latest release v0.5.0 to try it out.

Please let us know if there is any issue with the new identity type by opening a new issue.

Thank you!

**Comment by zhuolun-citrix** [comment]
Addressed in #36
